tests/mock_crust/utils.rs

```diff
@@ -329,51 +329,78 @@ pub fn create_connected_nodes_with_cache(network: &Network,
     nodes
 }
 
-// This creates new nodes (all with `use_cache` set to `true`) until the specified number of unique
-// split events have happened.  `split_count` should be at least 1.  Note that the final shape of
-// the network is still random, e.g. if `split_count` is 5, then the network could comprise groups
-// [0, 10, 110, 111] or [00, 01, 10, 11].
+// This creates new nodes (all with `use_cache` set to `true`) until the specified disjoint groups
+// have formed.
+//
+// `prefix_lengths` is an array representing the required `bit_count`s of the group prefixes.  For
+// example passing [1, 2, 3, 3] could yield a network comprising groups [0, 100, 101, 11], or
+// passing [2, 2, 3, 3, 3, 3] could yield [000, 001, 01, 100, 101, 11], while passing [1, 1] will
+// always yield groups [0, 1].
+//
+// The array is sanity checked (e.g. it would be an error to pass [1, 1, 1]), must comprise at
+// least two elements, and every element must be no more than `8`.
 pub fn create_connected_nodes_with_cache_until_split(network: &Network,
-                                                     split_count: usize)
+                                                     mut prefix_lengths: Vec<usize>)
```

Comment:
I did consider that approach, but wanted to make it easy to randomise the actual prefixes used.  I realise there _shouldn't_ be any difference between testing with [`0`, `10`, `11`] and [`00`, `01`, `1`], but I'd rather not make that assumption!

I guess it would be trivial to make a new function which does take a `Vec<Prefix>` and have `create_connected_nodes_with_cache_until_split()` call that after generating the random prefixes?